File: packages/implementations/OneLang-Regex-v0.1/StdLibs/one.pl
package OneRegex;

sub match_from_index
{
    my ($pattern, $input, $offset) = @_;
    pos($input) = $offset;
    my @matches = ($input =~ /\G$pattern/g);
    my $result = scalar(@matches) ? \@matches : undef;
    return $result;
}

1;

File: packages/implementations/OneLang-Regex-v0.1/StdLibs/one.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text.RegularExpressions;

public class OneRegex
{
    public static string[] MatchFromIndex(string pattern, string input, int offset)
    {
        var match = new Regex($"\\G{pattern}").Match(input, offset);
        if (!match.Success) return null;
        var result = match.Groups.Cast<Group>().Select(g => g.Value).ToArray();
        return result;
    }
}
